python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,72 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
```

Comment:
One thing that's probably worth mentioning is that this method is convenient for small groups. But if a group is too large (cannot fit on a single node), it won't work since it loads the entire group to memory. I imagine that people still need to use aggregate + map_batches in those cases (e.g. normalize a column).

Reply:
We chatted about this in a meeting about the desired API, +1 to making a note of this.

Reply:
For anything that can be achieved with aggregate (min/max/sum/std ...), essentially those ops that are associative/commutative, they should use those aggregate APIs: they are simpler and more efficient.

I put a comment at map_groups() here to call out this caveat (similar to pandas groupby().apply(): https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.core.groupby.GroupBy.apply.html)